src/Learning/KWData/KWMTDatabase.cpp

```diff
@@ -332,40 +345,94 @@ void KWMTDatabase::UpdateMultiTableMappings()
 			// Creation du mapping sauf si classe principale
 			if (referenceClass != mainClass)
 			{
-				// Creation d'un tableau de mapping, pour accuillir les mapping pour la classe externe
-				// en cours de traitement
-				oaCreatedMappings = new ObjectArray;
-				oaAllMainCreatedMappings.Add(oaCreatedMappings);
-
-				// Creation du mapping et memorisation de tous les mapping des sous-classes
+				// Premiere passe de creation du mapping de la table racine externe, avec des containers de travail
+				// Cela permet d'analyser la structure des mappings, sans impacter directement le contenu des containers globaux
+				odWorkingReferenceClasses.RemoveAll();
```

Comment:
Perhaps add a comment here, like:
```c++
// On s'assure que les containers soient vides
```
Or perhaps a more precise comment, which states that (unlike before) we make sure the data structures in question are only updated via mutation by `CreateMapping`.

Reply:
Amélioration du commentaire